include/aikido/control/Executor.hpp

```diff
@@ -0,0 +1,148 @@
+#ifndef AIKIDO_CONTROL_EXECUTOR_HPP_
+#define AIKIDO_CONTROL_EXECUTOR_HPP_
+
+#include <chrono>
+#include <future>
+#include <set>
+#include <vector>
+
+#include <dart/dynamics/DegreeOfFreedom.hpp>
+#include <dart/dynamics/Skeleton.hpp>
+
+#include "aikido/common/ExecutorThread.hpp"
+#include "aikido/common/pointers.hpp"
+
+namespace aikido {
+namespace control {
+
+AIKIDO_DECLARE_POINTERS(Executor)
+
+enum class ExecutorType
+{
+  kSTATE = -1,
+  kPOSITION = 0,
+  kVELOCITY = 1,
+  kEFFORT = 2,
+  kTRAJECTORY = 3,
+  kOTHER = 4
+};
+
+// inline function to get joint names from skeleton
+inline std::vector<std::string> skeletonToJointNames(
+    dart::dynamics::SkeletonPtr skeleton)
+{
+  std::vector<std::string> ret;
+  if (!skeleton)
+    return ret;
+  ret.reserve(skeleton->getNumDofs());
+  for (auto dof : skeleton->getDofs())
+  {
+    ret.push_back(dof->getName());
+  }
+  return ret;
+}
+
+// Parameter defaults
+constexpr std::chrono::milliseconds cDefaultThreadRate{10};
```

Comment:
I don't think we've previously used Hungarian notation for variable names, so we should be consistent.